src/common/mfu_daos.c

```diff
@@ -1244,6 +1309,90 @@ int daos_parse_epc_str(
     return 0;
 }
 
+#ifdef HDF5_SUPPORT
+static int serialize_daos_metadata(char *filename,
+                                   daos_args_t* da)
+{
+    int    rc = 0;
+    hid_t  status = 0;
+    struct hdf5_args hdf5;
+
+    rc = asprintf(&filename, "%s%s%s", da->daos_preserve_path, "/", "daos_metadata.hdf5");
+    printf("Writing metadata file: %s\n", filename);
+    if (rc == -1) {
+        MFU_LOG(MFU_LOG_ERR, "failed to write filename");
+        goto out;
+    }
+
+    /* TODO: much of this HDF5 setup should get moved into the HDF5 user
+     * interface */
+    hdf5.file = H5Fcreate(filename, H5F_ACC_TRUNC, H5P_DEFAULT, H5P_DEFAULT);
+    if (hdf5.file < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to create hdf5 file");
+        goto out;
+    }
```

Comment:
@dsikich I think we should set `rc = 1` here so the caller properly knows it failed.